.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,156 @@
+
+The `zdb` utility is the primary tool to debug ZFS. 
+
+Additionally, see 
+[build instruction](https://github.com/zfsonlinux/zfs/wiki/Building-ZFS)
+, there is `--enable-debug` flag, which can be set to enable all ASSERTs and additional correctness tests.
+
+### Where can I ask for help?
+The [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists) is the best place to ask for help.
+
+Please *note*, our issue tracker is **only for bugs**. 
+If you have any comments or suggestions regarding any enhancements, please use our 
+[mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+*Please* contact us via the [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists) if you aren't certain that you are experiencing a bug.
+
+If you run into an issue, please search our [issue tracker](https://github.com/zfsonlinux/zfs/issues) *first* to ensure the issue hasn't been reported before. 
+Open new issue only if you haven't found anything related to your case.
+
+You can open a new issue and search existing issues using the public
+[issue tracker](https://github.com/zfsonlinux/zfs/issues). 
+
+####When opening a new issue, please include the following information at the top of the issue:
+
+* What distribution (with version) you are using.
+* The spl and zfs versions you are using, installation method (repository or manual compilation).
+* Describe the issue you are experiencing.
+* Describe how to reproduce the issue.
+* Including any warning/errors/backtraces from the system logs.
+
+When a new issue is opened, it is not uncommon for developers to request additional information.
+
+In general, the more detail you share about a problem the quicker a developer 
+can resolve it. For example, providing a simple test case is always 
+exceptionally helpful.  
+
+Be prepared to work with the developers investigating your issue. 
+Your assistance is crucial in providing a quick solution.
+They may ask for information like:
+
+* Your pool configuration as reported by `zdb` or `zpool status`.
+* Your hardware configuration, such as
+  * Number of CPUs.
+  * Amount of memory.
+  * Whether your system has ECC memory.
+  * Whether it is running under a VMM/Hypervisor.
+  * Kernel version.
+  * Values of the spl/zfs module parameters.
+* Stack traces which may be logged to `dmesg`.
+
+####Issue template
+
+```
+**Which distribution are you using (version, kernel)**
+
+[Distribution, version, kernel version, architecture]
+
+**Which versions of spl/zfs packages are you using**
+
+[zfs version, spl version]
+
+**Describe the problem you're observing**
+
+[Describe the problem]
+
+**Describe how to reproduce the problem**
+
+1. [First Step]
+2. [Second Step]
+3. [Other Steps...]
+
+**Include any warning/errors/backtraces from the system logs**
+
+Wrap log text with ```
+
+[any log, see https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#reporting-bugs]
```

Comment:
I can add instructions to the issue template, but best way to do is as comments (so URL won't render). Or each user has to delete it from template each time.

Issue template section is from [guidance](https://mozillascience.github.io/working-open-workshop/contributing/), which @behlendorf proposed, and it is a good place to add more information and comments to it.